tests/bus_test.go

```diff
@@ -487,6 +487,161 @@ func TestSendingPanic(t *testing.T) {
 	}
 }
 
+func TestEmptyBody(t *testing.T) {
+	b := createNamedBusForTest(testSvc5)
+	proceed := make(chan bool)
+	b.SetGlobalRawMessageHandler(func(tx *sql.Tx, delivery *amqp.Delivery) error {
+		proceed <- true
+		return nil
+	})
+
+	err := b.Start()
+	if err != nil {
+		t.Errorf("could not start bus for test error: %s", err.Error())
+	}
+	defer func() {
+		err := b.Shutdown()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	conn, err := amqp.Dial(connStr)
+	if err != nil {
+		t.Error("couldnt connect to rabbitmq")
+	}
+	defer conn.Close()
+
+	ch, err := conn.Channel()
+	if err != nil {
+		t.Error("couldnt open rabbitmq channel for publishing")
+	}
+	defer ch.Close()
+
+	cmd := amqp.Publishing{}
+	err = ch.Publish("", testSvc5, true, false, cmd)
+	if err != nil {
+		t.Error("couldnt send message on rabbitmq channel")
+	}
+
+	select {
+	case <-proceed:
+		fmt.Println("success")
+	case <-time.After(2 * time.Second):
+		t.Fatal("timeout, failed to consume message with missing body")
+	}
+}
+
+func TestDeadEmptyBody(t *testing.T) {
+	b := createBusWithConfig(testSvc5, "grabbit-dead", true, true,
+		gbus.BusConfiguration{MaxRetryCount: 0, BaseRetryDuration: 0})
+
+	proceed := make(chan bool)
+	b.HandleDeadletter(func(tx *sql.Tx, delivery *amqp.Delivery) error {
+		proceed <- true
+		return nil
+	})
+
+	err := b.Start()
+	if err != nil {
+		t.Errorf("could not start bus for test error: %s", err.Error())
+	}
+
+	defer func() {
+		err := b.Shutdown()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	conn, err := amqp.Dial(connStr)
+	if err != nil {
+		t.Error("couldnt connect to rabbitmq")
+	}
+	defer conn.Close()
+	ch, err := conn.Channel()
+	if err != nil {
+		t.Error("couldnt open rabbitmq channel for publishing")
+	}
+	defer ch.Close()
+
+	headersMap := make(map[string]interface{})
+	headersMap["x-death"] = make([]interface{}, 0)
+	cmd := amqp.Publishing{Headers: headersMap}
+	err = ch.Publish("", testSvc5, true, false, cmd)
+	if err != nil {
+		t.Error("couldnt send message on rabbitmq channel")
+	}
+
+	select {
+	case <-proceed:
+		fmt.Println("success")
+	case <-time.After(2 * time.Second):
+		t.Fatal("timeout, failed to consume message with missing body")
+	}
+}
+
+func TestFailHandlerInvokeOfMessageWithEmptyBody(t *testing.T) {
```

Comment:
Since this test tests an edge case that is not easy to explain using only the test name I would add a comment explaining what this test actually tests.